tests/cnf/core/network/metallb/tests/bgp-multiservice.go

```diff
@@ -0,0 +1,153 @@
+	)
+
+	BeforeAll(func() {
+		By("Getting MetalLb load balancer ip addresses")
+		ipv4metalLbIPList, ipv6metalLbIPList, err = metallbenv.GetMetalLbIPByIPStack()
+		Expect(err).ToNot(HaveOccurred(), "An error occurred while "+
+			"determining the IP addresses from ECO_CNF_CORE_NET_MLB_ADDR_LIST environment variable.")
+
+		By("Selecting worker node for BGP tests")
+		cnfWorkerNodeList, err = nodes.List(APIClient,
+			metav1.ListOptions{LabelSelector: labels.Set(NetConfig.WorkerLabelMap).String()})
+		Expect(err).ToNot(HaveOccurred(), "Failed to discover worker nodes")
+		workerLabelMap, workerNodeList = setWorkerNodeListAndLabelForBfdTests(cnfWorkerNodeList, metalLbTestsLabel)
+		ipv4NodeAddrList, err = nodes.ListExternalIPv4Networks(
+			APIClient, metav1.ListOptions{LabelSelector: labels.Set(workerLabelMap).String()})
+		Expect(err).ToNot(HaveOccurred(), "Failed to collect external nodes ip addresses")
+
+		err = metallbenv.IsEnvVarMetalLbIPinNodeExtNetRange(ipv4NodeAddrList, ipv4metalLbIPList, nil)
+		Expect(err).ToNot(HaveOccurred(), "Failed to validate metalLb exported ip address")
+
+		By("Creating a new instance of MetalLB Speakers on workers")
+		err = metallbenv.CreateNewMetalLbDaemonSetAndWaitUntilItsRunning(tsparams.DefaultTimeout, workerLabelMap)
+		Expect(err).ToNot(HaveOccurred(), "Failed to recreate metalLb daemonset")
+	})
+
+	AfterAll(func() {
+		if len(cnfWorkerNodeList) > 2 {
+			removeNodeLabel(workerNodeList, metalLbTestsLabel)
+		}
+
+		By("Cleaning MetalLb operator namespace")
+		metalLbNs, err := namespace.Pull(APIClient, NetConfig.MlbOperatorNamespace)
+		Expect(err).ToNot(HaveOccurred(), "Failed to pull metalLb operator namespace")
+		err = metalLbNs.CleanObjects(
+			tsparams.DefaultTimeout,
+			metallb.GetBGPPeerGVR(),
+			metallb.GetBFDProfileGVR(),
+			metallb.GetBGPAdvertisementGVR(),
+			metallb.GetIPAddressPoolGVR(),
+			metallb.GetMetalLbIoGVR(),
+			metallb.GetFrrConfigurationGVR())
+		Expect(err).ToNot(HaveOccurred(), "Failed to remove object's from operator namespace")
+
+		By("Cleaning test namespace")
+		err = namespace.NewBuilder(APIClient, tsparams.TestNamespaceName).CleanObjects(
+			tsparams.DefaultTimeout,
+			pod.GetGVR(),
+			service.GetGVR(),
+			configmap.GetGVR(),
+			nad.GetGVR())
+		Expect(err).ToNot(HaveOccurred(), "Failed to clean test namespace")
+	})
+
+	It("Multi-Service Validation", reportxml.ID("47182"), func() {
+		By("should create an IPAddressPool and BGPAdvertisement for service 1")
+		ipAddressPool1 := setupBgpAdvertisementAndIPAddressPool(
+			tsparams.BGPAdvAndAddressPoolName, AddressPoolS1, netparam.IPSubnetInt32)
+
+		By("should create an IPAddressPool and BGPAdvertisement for service 2")
+		ipAddressPool2 := setupBgpAdvertisementAndIPAddressPool("bgp-test2", AddressPoolS2, netparam.IPSubnetInt32)
+
+		By("should create service 1 with 2 backend pods")
+		setupMetalLbService(tsparams.MetallbServiceName, netparam.IPV4Family, tsparams.LabelValue1, ipAddressPool1, "Cluster")
```

Comment:
Maybe we should add "cluster" to tsparam I see it hardcoded in a few of our test cases.